File: backend/src/main/java/com/healthchain/backend/model/user/Resource.java
package com.healthchain.backend.model.user;

import lombok.Data;

import java.util.List;

@Data
public class Resource {
    private String resourceType;
    private String id;
    private String gender;
    private List<ContactPoint> telecom;
    private List<Name> name;
    private String birthDate;
    private Reference managingOrganization; //hospName

    @Data
    public static class ContactPoint {
        private String system; //phone | fax | email | pager | url | sms | other
        private String value;
    }

    @Data
    public static class Name {
        private String use;
        private String family;
        private List<String> given;
    }

    @Data
    public static class Reference {
        private String identifier;
    }
}

//{
//        "resourceType": "Patient", //Patient / Practitioner
//        "id": "123456",
//        "gender": "male",
//        "telecom": [
//        {
//        "system": "phone",
//        "value": "555-1234"
//        },
//        {
//        "system": "email",
//        "value": "example@email.com"
//        }
//        ],
//        "name": [
//        {
//        "use": "official",
//        "family": "Doe",
//        "given": ["John", "M."]
//        }
//        ],
//        "birthDate": "1980-01-01"
//        }

File: network/custom-chaincode/chaincode/lib/model/Resource.js
'use strict';

class Resource {
    constructor(resourceType, id, gender, telecom, name, birthDate, managingOrganization) {
        this.resourceType = resourceType;
        this.id = id;
        this.gender = gender;
        this.telecom = telecom; //List<ContactPoint>
        this.name = name; //Name
        this.birthDate = birthDate;
        this.managingOrganization = managingOrganization; //Reference
    }
}

class ContactPoint {
    constructor(system, value) {
        this.system = system;
        this.value = value;
    }
}

class Name {
    constructor(use, family, given) {
        this.use = use;
        this.family = family;
        this.given = given;
    }
}

class Reference {
    constructor(identifier) {
        this.identifier = identifier;
    }
}

module.exports = {
    Resource,
    ContactPoint,
    Name,
    Reference
};